pkg/meta/autoid/autoid_service.go

```diff
@@ -190,10 +193,16 @@ func (b *backoffer) Backoff() {
 
 // ResetConn reset the AutoIDAllocClient and underlying grpc connection.
 // The next GetClient() call will recreate the client connecting to the correct leader by querying etcd.
-func (d *ClientDiscover) ResetConn(reason error) {
+func (d *ClientDiscover) ResetConn(version uint64, reason error) {
+	// Avoid repeated Reset operation
+	if !atomic.CompareAndSwapUint64(&d.version, version, version+1) {
```

Comment:
Introduce a `version` variable to avoid repeated `ResetConn()` and this also avoid a stale `ResetConn()` reset the newly created grpcConn mistakenly.